libethereum/Executive.cpp

```diff
@@ -306,19 +306,42 @@ bool Executive::call(CallParameters const& _p, u256 const& _gasPrice, Address co
 	return !m_ext;
 }
 
-bool Executive::create(Address _sender, u256 _endowment, u256 _gasPrice, u256 _gas, bytesConstRef _init, Address _origin)
+bool Executive::create(Address _txSender, u256 _endowment, u256 _gasPrice, u256 _gas, bytesConstRef _init, Address _origin)
+{
+	if (m_envInfo.number() >= m_sealEngine.chainParams().u256Param("metropolisForkBlock"))
+	{
+		m_newAddress = right160(sha3(MaxAddress.asBytes() + sha3(_init).asBytes()));
+		return executeCreate(_txSender, _endowment, _gasPrice, _gas, _init, _origin);
+	}
+	else
+		return createOpcode(_txSender, _endowment, _gasPrice, _gas, _init, _origin);
+}
+
+bool Executive::createOpcode(Address _txSender, u256 _endowment, u256 _gasPrice, u256 _gas, bytesConstRef _init, Address _origin)
```

Comment:
`txSender` sounds like the sender of the transaction, but I guess this address points to the contract executing `CREATE`.